File: Python/03-cuarto/03-script.py
def convertir_grados(grados=None, unidad=None):
    # Validación: Si no se ingresaron grados
    if grados is None:
        print("No ingresaste grados a convertir")
        return

    # Validación: Si grados no es un número
    if not isinstance(grados, (int, float)):
        print(f"El valor '{grados}' ingresado, NO es un número")
        return

    # Validación: Si no se ingresó la unidad
    if unidad is None:
        print("No ingresaste el tipo de unidad a convertir")
        return

    # Validación: Si unidad no es una cadena
    if not isinstance(unidad, str):
        print(f"El valor '{unidad}' ingresado, NO es una cadena de texto")
        return

    # Validación: Si unidad no es 'C' o 'F'
    if len(unidad) != 1 or unidad not in ('C', 'F'):
        print("Valor de unidad no reconocido")
        return

    # Conversión
    if unidad == "C":
        resultado = round(grados * (9 / 5) + 32)
        print(f"{grados}C = {resultado}F")
    else:
        resultado = round((grados - 32) * (5 / 9))
        print(f"{grados}F = {resultado}C")

# Ejemplos de prueba
convertir_grados()             # No ingresaste grados a convertir
convertir_grados("2")          # El valor '2' ingresado, NO es un número
convertir_grados(2)            # No ingresaste el tipo de unidad a convertir
convertir_grados(2, True)      # El valor 'True' ingresado, NO es una cadena de texto
convertir_grados(2, "hola")    # Valor de unidad no reconocido
convertir_grados(2, "E")       # Valor de unidad no reconocido
convertir_grados(0, "C")       # 0C = 32F
convertir_grados(32, "F")      # 32F = 0C
convertir_grados(100, "F")

File: Php/03-cuarto/03-script.php
<?php

function convertirGrados($grados = null, $unidad = null) {
    // Validación: Si no se ingresaron grados
    if (is_null($grados)) {
        echo "No ingresaste grados a convertir\n";
        return;
    }

    // Validación: Si unidad es null
    if (is_null($unidad)) {
        echo "No ingresaste el tipo de unidad a convertir\n";
        return;
    }

    // Validación: Si unidad no es "C" o "F"
    if (strlen($unidad) != 1 || !in_array($unidad, ["C", "F"])) {
        echo "Valor de unidad no reconocido\n";
        return;
    }

    // Conversión
    if ($unidad === "C") {
        $resultado = round($grados * (9 / 5) + 32);
        echo "{$grados}C = {$resultado}F\n";
    } else {
        $resultado = round(($grados - 32) * (5 / 9));
        echo "{$grados}F = {$resultado}C\n";
    }
}

// Ejemplos de prueba
convertirGrados();               // No ingresaste grados a convertir
convertirGrados("2");             // No ingresaste el tipo de unidad a convertir
convertirGrados(2, "hola");       // Valor de unidad no reconocido
convertirGrados(2, "E");          // Valor de unidad no reconocido
convertirGrados(0, "C");          // 0C = 32F
convertirGrados(32, "F");         // 32F = 0C
convertirGrados(100, "F");        // 100F = 38C

?>